arrocke/gloss-translation
# refactor: track changes to phrases and glosses

## What has changed

* phrases track the time of creation and deletion, as well as who created and deleted the phrase
* all queries filter out deleted queries - we keep them so we can restore them if there were deleted on accident, and so we can see who deleted
* gloss history entries were renamed to gloss events and now associate with the phrase rather than a specific word

I ended up deviating a bit from the initial design because it made things a littler simpler to implement. Because we don't need to modify the words in a phrase once it is created, we can just track the created and deleted date on the phrase itself. That negates the need for a more complicated event system for glosses since the current system will work just fine with phrase IDs rather than word and language IDs



## Connected Issues

closes #384 



## QA Steps

1. Run `nx run db:prisma migrate dev` to update your database with the new schema
2. Try updating a few glosses to see what happens as phrases and gloss events are created



## Post-Deployment
